frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -1,25 +1,44 @@
 import { type ReactNode } from 'react';
-import { Navigate } from 'react-router-dom';
+import { Navigate, useLocation } from 'react-router-dom';
```

Comment:
Re-export useLocation in #libs/hooks/hooks.ts
Also re-export Navigate from #libs/components/components.ts